service/history/historyEngine.go

```diff
@@ -1151,9 +1156,12 @@ Update_History_Loop:
 					continue Process_Decision_Loop
 				}
 				attributes := d.CompleteWorkflowExecutionDecisionAttributes
-				if err = validateCompleteWorkflowExecutionAttributes(attributes); err != nil {
+				if err = validateCompleteWorkflowExecutionAttributes(attributes, sizeLimitWarn, sizeLimitError,
+					domainID, workflowExecution.GetWorkflowId(), workflowExecution.GetRunId(),
+					e.metricsClient, metrics.HistoryRespondDecisionTaskCompletedScope, e.logger); err != nil {
```

Comment:
I think it is better to fail the workflow instead of failing decision.  Otherwise client would just keep on retrying with the same payload.  Once we have restart user can always restart failed workflows automatically.

Reply:
only for this decision DecisionTypeCompleteWorkflowExecution? Or for all other decisions?